src/AzureClient/Mocks/MockCloudJob.cs

```diff
@@ -7,9 +7,89 @@
 using Azure.Quantum.Jobs.Models;
 using System;
 using System.IO;
+using System.Collections.Generic;
 
 namespace Microsoft.Quantum.IQSharp.AzureClient
 {
+    public class MockCostEstimate
+    {
+        public MockCostEstimate(string currencyCode, IReadOnlyList<UsageEvent> events, float? estimatedTotal)
+        {
+            this.InternalCostEstimate = MockHelper.CreateWithNonPublicConstructor<CostEstimate>(currencyCode, events, estimatedTotal);
+        }
+
+        public string CurrencyCode
+        {
+            get => this.InternalCostEstimate.CurrencyCode;
+            set => this.InternalCostEstimate.SetReadOnlyProperty<CostEstimate>(nameof(CurrencyCode), value);
```

Comment:
Minor code style nit, but can `<CostEstimate>` elided here? It looks like that's the receiver type for an extension method, which normally can be inferred as a type parameter.

Reply:
In this case it cannot be inferred because it would pick this same implementation of the setter and create a infinite recursion. We want to pick the implementation of the base class.